pro_wes/tasks/track_run_progress.py

```diff
@@ -106,7 +106,13 @@ def task__track_run_progress(
     while not run_state.is_finished:
         sleep(controller_config.polling_wait)
         try:
-            response = wes_client.get_run_status(
+            if document.wes_endpoint is None:
+                raise ValueError("WES run ID not available.")
+
+            if document.wes_endpoint.run_id is None:
+                raise ValueError("WES run ID not available.")
+
+            wes_client.get_run_status(
```

Comment:
Check other comments about custom errors and possibly using another approach to validate `.wes_endpoint`